### PR TITLE
Milestone 6. People & Groups

### PR DESCRIPTION
**Pull request by BamblooV**
1. Task: [link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/connections/milestone_6.people_groups.md)
2. Screenshot: 
chrome_0AiOZajiCY

3. Deploy: [link](https://bambloov.github.io/finatl-task/)
4. Done 13.12.2023 / deadline 20.12.2023
5. Score: 175 / 175 

## Комментарии
Куча бойлерплейта. Мб где-то по неопытности накостылил и намудрил

### Profit

- [x] default page for authorized user: **10 points**
- [x] page is divided on 2 vertical sections with independent content: **5 points**

#### Group section (left)

- [x] the list of available groups is loaded if user opens this page first time: **5 points**
- [x] the list item created by current user should contain _Delete_ button: **10 points**
- [x] the confirmation modal appears after clicking on _Delete_ button on list item with _Cancel_,
  _Delete_ button inside. If user clicks _Cancel_ the modal disappears. If user clicks _Delete_ the
  http-request is sent and item is removed from the list after succeeded response: **15 points**
- [x] clicking on _Update_ button sends corresponding http-request and update group
  list if succeeded: **10 points**
- [x] countdown appears for 1 minute after clicking on _Update_ button
  (except if error occurs): **10 points**
- [x] _Update_ button is disabled after clicking during updating and until the timer
  is active: **5 points**
- [x] clicking on _Create_ button the modal window is opened. There is form with validation and
  submit button: **10 points**
- [x] submit button in modal window should be disabled until form is valid: **5 points**
- [x] clicking on submit button in modal window the appropriate http-request is sent to create new
  group. Modal window is closed only if http-request succeeded: **15 points**
- [x] toast messages with appropriate text are displayed if http-request fails or
  succeed: **10 points**
- [x] clicking on list item the user is redirected to group dialog page: **5 points**

#### People list (right)

- [x] the list of people is loaded if user opens this page first time: **10 points**
- [x] the list item with which current user already has active conversation has
  special background: **10 points**
- [x] clicking on _Update_ button sends corresponding http-request and update people list
  if succeeded: **10 points**
- [x] countdown appears for 1 minute after clicking on _Update_ button
  (except if error occurs): **10 points**
- [x] _Update_ button is disabled after clicking during updating and until the
  timer is active: **5 points**
- [x] clicking on list item the user is redirected to personal conversation page. New conversation (via
  certain http-request) is created if it has not already created before transition: **15 points**

#### Group section

- [ ] list of groups via `/groups/list` is automatically loaded more than once during 1 browser
  session (until the user refreshes the page) if user do not click _Update_ button. For instance,
  when user navigates through the pages, sends new messages, deletes or creates
  group(s): **-30 points**

#### People list

- [ ] list of conversations via `/conversations/list` is automatically loaded more than once during 1
  browser session (until the user refreshes the page) if user do not click _Update_ button. For
  instance, when user navigates through the pages, sends new messages, deletes or creates
  conversation(s): **-20 points**
- [ ] list of users via `/users` is automatically loaded more than once during 1 browser session (until
  the user refreshes the page) if user do not click _Update_ button. For instance, when user
  navigates through the pages, sends new messages, deletes or creates
  conversation(s): **-20 points**

### CODE REVIEW

File: src/app/messenger/components/create-group-modal/create-group-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }}</small>
+  </div>
+  <ng-template pTemplate="footer">
+    <p-button
+      [disabled]="control.invalid"
+      icon="pi pi-check"
+      (click)="onSubmit()"
+      label="Ok"
+      styleClass="p-button-text"
+      [disabled]="!!(loading$ | async)"></p-button>
```
Comment (DevArslan) [inline src/app/messenger/components/create-group-modal/create-group-modal.component.html:18]: Два раза [disabled] прокидывается
```
 [disabled]="control.invalid"
 [disabled]="!!(loading$ | async)"
```

File: src/app/messenger/services/chats-api.service.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  private baseUrl = environment.baseApiUrl;
+
+  fetchGroups(): Observable<{ groups: Group[]; count: number }> {
+    return this.store.select(GroupsSelectors.selectGroupsAndCount).pipe(
+      take(1),
+      catchError(() => {
+        return throwError(() => ({ type: 'Unknown', message: 'NGRX error.' }));
+      }),
+      switchMap(([groups, count]) => {
+        if (!groups || !groups.length) {
```
Comment (DevArslan) [inline src/app/messenger/services/chats-api.service.ts:31]: Предложение
`if (!groups?.length)`

File: src/app/messenger/components/create-group-modal/create-group-modal.component.html (hunk 1 above)
Comment (BamblooV) [reply src/app/messenger/components/create-group-modal/create-group-modal.component.html:18]: fixed

File: src/app/messenger/services/chats-api.service.ts (hunk 2 above)
Comment (BamblooV) [reply src/app/messenger/services/chats-api.service.ts]: refactored

**Pull request merged**